### PR TITLE
Card

### PR DESCRIPTION
**Pull request by partha8**
## Files to check
- [src/components/Card/Card.js](https://github.com/partha8/melonn/compare/card?expand=1#diff-c140b1cd54cc14c0d77d8af6d1b69d4a1e5456147d4379c8fcb3629e651d0392)
- [src/components/Card/Colors/Colors.js](https://github.com/partha8/melonn/compare/card?expand=1#diff-4ef3bec4ce09027302018b509b36da5adc0ea8fc9d9fe9ac94cc9d653b180324)
- [src/components/Card/Colors/colors.module.css](https://github.com/partha8/melonn/compare/card?expand=1#diff-d2ff421dca67b7bc3bfae98ca8b8afb908ba428e5f167b12dbc42e8790ef3d88)
- [src/components/Card/card.module.css](https://github.com/partha8/melonn/compare/card?expand=1#diff-84710932c9a04ea0dfb536ca3074743b2f9f77e3246c4d997e02d029a4dad208)

## preview
color-palette

### CODE REVIEW

File: src/components/Card/Card.js [hunk 1]
```diff
@@ -1,32 +1,75 @@
...
         <article>
           <h6>{title}</h6>
           <p>{body}</p>
         </article>
         <span className="icon">{pinned ? <BsFillPinFill /> : <BsPin />}</span>
       </section>
+      <section className={styles.tags}>
+        {tags.map((tag, index) => {
+          return (
+            <p key={index} className={styles.tag}>
```
Comment (foolhardy21) [inline src/components/Card/Card.js:57]: one suggestion would be to replace `index` with an `id` for the `key`.

File: src/components/Card/Colors/Colors.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const [showPalette, setShowPalette] = useState(false);
+
+  const updateColor = async (id, color) => {
+    const docRef = doc(db, "notes", id);
+    await updateDoc(docRef, {
+      color: color,
+    });
+  };
+
+  const clickHandler = (e) => {
```
Comment (foolhardy21) [inline src/components/Card/Colors/Colors.js]: another suggestion would be to name this handler something more specific.

File: src/components/Card/Colors/colors.module.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
+.palette {
+    position: absolute;
+    background: #fff;
+    /* border: 1px solid black; */
```
Comment (foolhardy21) [inline src/components/Card/Colors/colors.module.css:4]: you can remove this commented code if not required.

**Review by foolhardy21 — APPROVED** [review]
the UI looks good. have given some minor suggestion in code. Other than it is good to go ahead.

File: src/components/Card/Card.js (hunk 1 above)
Comment (partha8) [reply src/components/Card/Card.js:57]: in the firestore, I am storing this ``tags`` array in every doc, and referring to an id would be tough. I am looking for a way around though. Thank you for the review.

File: src/components/Card/Colors/colors.module.css (hunk 3 above)
Comment (partha8) [reply src/components/Card/Colors/colors.module.css:4]: Yup, i missed it. Thank you!

File: src/components/Card/Colors/Colors.js (hunk 2 above)
Comment (partha8) [reply src/components/Card/Colors/Colors.js]: done!

File: src/hooks/useClickOutside.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+export const useClickOutside = (handler) => {
+  const domNode = useRef();
+  useEffect(() => {
+    const clickHandler = (e) => {
+      if (domNode.current && !domNode.current.contains(e.target)) {
+        handler();
+      }
+    };
+    document.addEventListener("mousedown", clickHandler);
```
Comment (VLeads) [inline src/hooks/useClickOutside.js:11]: In react, we normally not mix JS syntax. If there is any alternative, you should use that.

**Review by VLeads — APPROVED** [review]
Code looks good. Good to go !!

File: src/hooks/useClickOutside.js (hunk 4 above)
Comment (partha8) [reply src/hooks/useClickOutside.js:11]: This is a different usecase where I was required to access dom directly. I have looked for solutions and this was used in multiple places. Thank you for the review!

**Pull request merged**